cmd/fossa/update.go

```diff
@@ -0,0 +1,49 @@
+package main
+
+import (
+	"fmt"
+	"github.com/blang/semver"
+	logging "github.com/op/go-logging"
+	"github.com/rhysd/go-github-selfupdate/selfupdate"
+	"github.com/urfave/cli"
+)
+
+var updateLogger = logging.MustGetLogger("update")
+var updateEndpoint := "fossas/fossa-cli"
+
+func checkUpdate() error {
+	latest, found, err := selfupdate.DetectLatest(updateEndpoint)
+	if err != nil {
+		return err // unable to update due to GH error
+	}
+
+	v := semver.MustParse(version)
+	if !found || latest.Version.Equals(v) {
+		return fmt.Errorf("current binary is latest version") 
+	}
+}
+
+func doSelfUpdate() error {
+	v := semver.MustParse(version)
+	latest, err := selfupdate.UpdateSelf(v, updateEndpoint)
+	if err != nil { 
+		return err
+	}
+	if latest.Version.Equals(v) {
+		// latest version is the same as current version. It means current binary is up to date.
+		return fmt.Errorf("no update required; currently on latest")
+	}
+	return nil
+}
+
+func updateCmd(c *cli.Context) {
+	if err := checkUpdate(); err != nil {
+		updateLogger.Fatalf("unable to update (%s)", err)
```

Comment:
_Logged_ messages (distinct from _error messages_) should generally be `Sentence case`. The reason why error messages should _not_ be `Sentence case` is because they're usually embedded within a sentence when they're finally logged to display to the user (e.g. `logger.Fatalf("Problem happened: %s", err.Error())`).